src/leiningen/eastwood.clj

```diff
@@ -84,39 +83,62 @@ http://dev.clojure.org/jira/browse/CLJ-1445"
+
+(defn- add-eastwood [project]
+  (let [eastwood-dep (->> project
+                          :plugins
+                          (filter (comp (partial = 'jonase/eastwood) first))
+                          first)]
+    (update project :dependencies (comp vec conj) eastwood-dep)))
+
+;; The rationale for this function is as follows:
+;; In order for this code to be run, eastwood has to be present in some
+;; :plugin - vector, either in a `project.clj` or in `~/.lein/profiles.clj`.
+;; The Eastwood readme does not mandate that you also put eastwood in the
+;; `:dependencies` vector if you put it in a `~/.lein/profiles.clj` `:plugin`
+;; vector. If this is the case, we need to manually add it here:
+;; So being here, we know at least htat we have eastwood defined as a plugin.
+
+(defn- maybe-add-eastwood [project]
+  (cond-> project
+    (missing-eastwood-dependency? project) add-eastwood))
 
 (defn eastwood
   ([project] (eastwood project "{}"))
   ([project opts]
-     (cond
-      (= opts "help") (println (help))
-
-      (= opts "lein-project")
-      (do
-        (pprint-meta (into (sorted-map) project))
-        (println "\nValue of :eastwood key in project map:")
-        (pprint-meta (into (sorted-map) (:eastwood project))))
-
-      :else
-      (let [leiningen-paths (select-keys project [:source-paths
-                                                  :test-paths])
-            leiningen-opts (:eastwood project)
-            cmdline-opts (read-string opts)
-            opts (merge leiningen-paths leiningen-opts cmdline-opts)]
-        (when (some #{:options} (:debug opts))
-          (println "\nLeiningen paths:")
-          (pprint-meta (into (sorted-map) leiningen-paths))
-          (println "\nLeiningen options map:")
-          (pprint-meta (into (sorted-map) leiningen-opts))
-          (println "\nCommand line options map:")
-          (pprint-meta (into (sorted-map) cmdline-opts))
-          (println "\nMerged options map:")
-          (pprint-meta (into (sorted-map) opts))
-          (println "\nLeininge project map:")
-          (pprint-meta (into (sorted-map) project))
-          (println)
-          (flush))
-        (leval/eval-in-project
-         (add-if-missing project ['jonase/eastwood eastwood-version-string])
-         `(eastwood.versioncheck/run-eastwood '~opts)
-         '(require 'eastwood.versioncheck))))))
+   (cond
+     (= opts "help") (println (help))
+     (= opts "lein-project")
+     (do
+       (pprint-meta (into (sorted-map) project))
+       (println "\nValue of :eastwood key in project map:")
+       (pprint-meta (into (sorted-map) (:eastwood project))))
+
+     :else
+     (let [leiningen-paths (select-keys project [:source-paths
+                                                 :test-paths])
+           leiningen-opts (:eastwood project)
+           cmdline-opts (read-string opts)
+           opts (merge leiningen-paths leiningen-opts cmdline-opts)
+           debug? (some #{:options} (:debug opts))]
```

Comment:
Not a big deal, but perhaps this should follow the Leiningen plugin writing advice here 
https://github.com/technomancy/leiningen/blob/master/doc/PLUGINS.md#writing-a-plugin

which, I'll quote the relevant part:
> When emitting output, please use leiningen.core.main/info, leiningen.core.main/warn, and leiningen.core.main/debug rather than println since these will respect the user's output settings.